aws/data_source_aws_prefix_list_test.go

```diff
@@ -98,3 +99,47 @@ data "aws_prefix_list" "s3_by_id" {
   }
 }
 `
+
+func TestAccDataSourceAwsPrefixList_matchesTooMany(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsPrefixListConfig_matchesTooMany,
+				ExpectError: regexp.MustCompile(`more than one prefix list matched the given set of criteria`),
+			},
+		},
+	})
+}
+
+const testAccDataSourceAwsPrefixListConfig_matchesTooMany = `
+data "aws_prefix_list" "test" {}
+`
+
+func TestAccDataSourceAwsPrefixList_nameDoesNotOverrideFilter(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// The vanilla DescribePrefixLists API only supports filtering by
+				// id and name. In this case, the `name` attribute and `prefix-list-id`
+				// filter have been set up such that they conflict, thus proving
+				// that both criteria took effect.
+				Config:      testAccDataSourceAwsPrefixListConfig_nameDoesNotOverrideFilter,
+				ExpectError: regexp.MustCompile(`no matching prefix list found`),
+			},
+		},
+	})
+}
+
+const testAccDataSourceAwsPrefixListConfig_nameDoesNotOverrideFilter = `
+data "aws_prefix_list" "test" {
+  name = "com.amazonaws.us-west-2.s3" 
```

Comment:
Some of our newer codebase linting will likely pick this up on rebase/merge for not being AWS Region agnostic. Similar to  the website example you added (👍 ) we should use the `aws_region` data source here so the configuration can be applied in anywhere. This also means we'll need to remove the hardcoded EC2 Prefix List identifier. Something like this should get you close:

```terraform
data "aws_region" "current" {}

data "aws_prefix_list" "dynamodb" {
  name = "com.amazonaws.${data.aws_region.current.name}.dynamodb"
}

data "aws_prefix_list" "test" {
  name = "com.amazonaws.${data.aws_region.current.name}.s3"

  filter {
    name = "prefix-list-id"
    values = [data.aws_prefix_list.dynamodb.id]
  }
}
```